dask/array/routines.py

```diff
@@ -2046,7 +2045,7 @@ def coarsen(reduction, x, axes, trim_excess=False, **kwargs):
         msg = f"Coarsening factors {axes} do not align with array shape {x.shape}."
         raise ValueError(msg)
 
-    if "dask" in inspect.getfile(reduction):
+    if "dask" in reduction.__module__:
```

Comment:
We might want to make this even more restrictive with something like:

```suggestion
    if reduction.__module__.startswith("dask."):
```

What do you think?

Reply:
Hm, yes, I think restricting it further is a good suggestion. Thanks @jrbourbeau